### ISSUE TITLE
[Feature Request] Cross plugin task support

### ISSUE THREAD

**Issue by roovo** [roovo/obsidian-card-board#92]
Add support for reading due and completed dates in tasks defined using either Tasks, Dataview, Reminders, or Kanban format.

```
## Tasks
- [ ] Task 📅 2021-09-16 ✅ 2021-04-24
- [x] Task ⏳ 2021-09-16                (UPDATED: ignoring scheduled dates for now)

## Dataview
- [ ] Task due:: 2021-09-16 completion:: 2021-09-16

## Reminder
- [ ] Task (@2021-09-16 20:40)

## Kanban
- [ ] Task @{2021-09-16} @@{20:26}
- [ ] Task @[[2021-09-16]] @@{20:26}

```

- [x] Tasks
- [x] Dataview
- [ ] Reminders
- [ ] Kanban

For dataview, if the plugin is installed I should use the values in the dataview settings (otherwise the defaults - due:: and completion:: ).

[UPDATED see #94 - for now only considering Due dates] For Tasks, there are [due, scheduled, and start dates](https://obsidian-tasks-group.github.io/obsidian-tasks/getting-started/dates/).  Looks like the most appropriate interpretation is to use the scheduled date if present (and show as scheduled on the card), if not use the Due date.  I don't think Start dates are relevant for CardBoard.

Provide an option to choose which format to use when marking a task as complete.

This is a superset of what was asked for in #77

**Comment by roovo** [comment]
For Tasks compatibility (for now at least - see #94) I will only handle due dates (specified with 📅) as these appear to the the default and most raised.  Adding other date concepts has wider implications: which date should determine the column on date boards, which date should be displayed on cards, which date should change on (not yet implemented) drag-and-drop on date boards?

**Comment by roovo** [comment]
closed #96 as Tasks compatibility just released in v 0.6.0

**Comment by roovo** [comment]
closing this FR:

Reminder compatibility moved to [Handle dates other than due date](https://github.com/roovo/obsidian-card-board/issues/94#issuecomment-1326787533).

Kanban compatibility: at the moment not planned as Kanban is already a kanban, so using it in conjunction with Cardboard is probably of limited use.  Also there are some fundamental differences that would make it hard to ensure consistent cross-plugin behavior (e.g. marking a task checkbox as done in Cardboard always moves it to the Done column on Cardboard but this is not how Kanban works,  Cardboard treats all following indented lines as sub-tasks and/or description whereas Kanban expects everything to be on the one line).

Tasks: re-opened #26 (on handling task recurrence).  Also #94 covers the use of other date types.
